provider/datagen/src/baked_exporter.rs

```diff
@@ -2,20 +2,61 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use databake::{quote, CrateEnv, TokenStream};
+//! A data exporter that bakes the data into Rust code.
+//!
+//! This module can be used as a target for the `icu_datagen` crate.
+//!
+//! # Examples
+//!
+//! ```
+//! use icu_datagen::prelude::*;
+//! use icu_provider::hello_world::*;
+//! use icu_datagen::baked_exporter::*;
+//!
+//! let demo_path = std::env::temp_dir().join("icu4x_baked_demo");
+//! # let _ = std::fs::remove_dir_all(&demo_path);
+//!
+//! // Set up the exporter
+//! let mut exporter = BakedExporter::new(demo_path.clone(), Default::default()).unwrap();
+//!
+//! // Export something
+//! DatagenProvider::default()
+//!     .export(
+//!         [HelloWorldV1Marker::KEY].into_iter().collect(),
+//!         exporter
+//!     ).unwrap();
+//! #
+//! # let _ = std::fs::remove_dir_all(&demo_path);
+//! ```
+//!
+//! The resulting module structure can now be used like this:
+//!
+//! ```compile_fail
+//! use icu_provider::prelude::*;
+//!
+//! struct MyDataProvider;
+//!
+//! mod baked {
+//!   include!("/path/to/mod/")
+//!   impl_data_provider!(MyDataProvider);
+//! }
```

Comment:
Issue: Fix missing semicolon and indentation.

Request (optional): It would be nice if this example could compile. You could potentially do this by pointing at a path that contains real hello-world bake data.